### ISSUE TITLE
Problems with the input path pattern matching of nested folders

### ISSUE THREAD

**Issue by indieSoftware** [scribd/Weaver#98]
I have added weaver 0.12 via pods to my project (thanks for the pod support). 

My project's root folder where my project and workspace files are located contain a folder "Folder1" (which represents here my app's source code folder) and another folder "Folder2" (which represents a Unit Test folder, a UI Test folder, a widget, etc). "Folder1" then contains subfolders ("Folder1a" and "Folder1b") with some swift files ("Source1" and "Source2"). "Folder1b" also contains another subfolder "Subfolder1b" with additional source files ("Source3").

Therefore the project's folder looks similar to:

┣Folder1
┃┣Folder1a
┃┃┣Source1.swift
┃┣Folder1b
┃┃┣Source2.swift
┃┃┣Subolder1b
┃┃┃┣Source3.swift
┣Folder2
┃┣Source4.swift
┣.xcodeproj
┣.xcworkspace

I'm now located in my project's root folder and I run weaver as recommended in the readme the following way (the call `Pods/WeaverDI/weaver/bin/weaver` is for simplicity just replaced with `weaver`):

`weaver swift --project-path Folder1 --input-path "*.swift" --input-path "**/*.swift" --output-path SomeFolder`

That results in weaver saying `Unknown Arguments: Folder2/Source4.swift` which is really strange because I expect weaver only looks inside of Folder1, not Folder2, at least that's how I interpret the parameter `project-path`.

So I tried the following;

`weaver swift --project-path . --input-path "Folder1/**/*.swift" --output-path SomeFolder`

That worked, the injection began, but only the following files were parsed:

```
Parsing...
<- 'Folder1/Folder1a/Source1.swift'
<- 'Folder1/Folder1b/Source2.swift'
```

What with `Source3.swift` in the `Subfolder1b`? It seems the pattern `**` doesn't contain sub-sub-folders and thus doesn't match to `Folder1b/Subfolder1b`.

Btw, double nesting with `**/**` doesn't seem to work either:

`weaver swift --project-path . --input-path "Folder1/**/**/*.swift" --output-path SomeFolder`

Result: 
`Unknown Arguments: Folder1/Folder1b/Subfolder1b/Source3.swift` 

Am I doing something wrong?
However, a call like the following works again (imagine Subfolder1b has another subfolder in it):

`weaver swift --project-path Folder1 --input-path "Folder1b/**/**/*.swift" --output-path SomeFolder`

Is it possible to make it match recursively, everything what is in a given folder regardless how deep it is nested? If not, then I think that's a big issue, because I don't want to modify the script every time I add a new subfolder to my project.

So IMHO to match all swift files within `Folder1` a call like the following would be nice and sufficient:

`weaver swift --project-path Folder1 --output-path SomeFolder`

No need to address `*.swift` files or can weaver handle other file formats, too? If so, then maybe a call like:

`weaver swift --project-path Folder1 --input-path "*.swift" --output-path SomeFolder`

Here the `input-path` parameter is executed on each subfolder of `project-path` so there would be no need to add a parameter like `**/*.swift`. If there are scenarios where a recursive processing is not wished, then maybe add a new `--recursive-off` parameter and / or work with `ignored-path` arguments. However, I think the default use case would be to parse all swift files in the project beginning with a specific source root folder and maybe only excluding some special files / folders.

**Comment by trupin** [comment]
It looks like `bash` (which I assume you're using) uses its own globing when writing `"*.swift"`. I thought it was doing so only when writing `*.swift` without the double or single quotes.

Weaver can't work if globing is done upfront since `--input-path` can only take one value at a time.

This is a very annoying bug which confirms my doubts concerning this whole globing feature that I added in this release.

> Here the `input-path` parameter is executed on each subfolder of `project-path` so there would be no need to add a parameter like `**/*.swift`. If there are scenarios where a recursive processing is not wished, then maybe add a new `--recursive-off` parameter and / or work with `ignored-path` arguments. However, I think the default use case would be to parse all swift files in the project beginning with a specific source root folder and maybe only excluding some special files / folders.

I like this idea and I think this is indeed the way to go.

**Comment by trupin** [comment]
Should be fixed with #100
